lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -20,110 +20,86 @@ func New(betRepository BetRepository) *Handler {
 				continue
 			}
-
-			// Calculate the resulting bet, which should be published.
-			resultingBet := rabbitmqmodels.Bet{
-				Id:                   domainBet.Id,
-				CustomerId:           domainBet.CustomerId,
-				Status:               domainBet.Status,
-				SelectionId:          domainBet.SelectionId,
-				SelectionCoefficient: domainBet.SelectionCoefficient,
-				Payment:              domainBet.Payment,
-			}
-
-			select {
-			case resultingBets <- resultingBet:
-			case <-ctx.Done():
-				return
-			}
 		}
 	}()
 
-	return resultingBets
+	return betsCalculated
 }
 
-// HandleBetsCalculated handles bets calculated.
-func (h *Handler) HandleBetsCalculated(
+// HandleEventUpdates handles updated bets.
+func (h *Handler) HandleEventUpdates(
 	ctx context.Context,
-	betsCalculated <-chan rabbitmqmodels.BetCalculated,
-) <-chan rabbitmqmodels.Bet {
-	resultingBets := make(chan rabbitmqmodels.Bet)
+	eventUpdates <-chan rabbitmqmodels.BetEventUpdate,
+) <-chan rabbitmqmodels.BetCalculated {
+	betsCalculated := make(chan rabbitmqmodels.BetCalculated)
 
 	go func() {
-		defer close(resultingBets)
+		defer close(betsCalculated)
 
-		for betCalculated := range betsCalculated {
-			log.Println("Processing bet calculated, betId:", betCalculated.Id)
+		for eventUpdate := range eventUpdates {
+			log.Println("Processing event updates, betId:", eventUpdate.Id)
 
 			// Fetch the domain bet.
-			domainBet, exists, err := h.betRepository.GetBetByID(ctx, betCalculated.Id)
+			domainBets, exists, err := h.betRepository.GetBetsBySelectionID(ctx, eventUpdate.Id)
 			if err != nil {
 				log.Println("Failed to fetch a bet which should be updated, error: ", err)
 				continue
 			}
 			if !exists {
-				log.Println("A bet which should be updated does not exist, betId: ", betCalculated.Id)
+				log.Println("A bet which should be updated does not exist, betId: ", eventUpdate.Id)
 				continue
 			}
 
-			// Update the domain bet based on incoming changes.
-			domainBet.Status = betCalculated.Status
-			domainBet.Payout = betCalculated.Payout
-
-			// Update the domain bet into the repository.
-			err = h.betRepository.UpdateBet(ctx, domainBet)
-			if err != nil {
-				log.Println("Failed to update bet, error: ", err)
-				continue
-			}
-
-			// Calculate the resulting bet, which should be published.
-			resultingBet := rabbitmqmodels.Bet{
-				Id:                   domainBet.Id,
-				CustomerId:           domainBet.CustomerId,
-				Status:               domainBet.Status,
-				SelectionId:          domainBet.SelectionId,
-				SelectionCoefficient: domainBet.SelectionCoefficient,
-				Payment:              domainBet.Payment,
-				Payout:               domainBet.Payout,
-			}
-
-			select {
-			case resultingBets <- resultingBet:
-			case <-ctx.Done():
-				return
+			for _, domainBet := range domainBets {
+				// Calculate the resulting bet, which should be published.
+				betCalculated := rabbitmqmodels.BetCalculated{
+					Id:                   domainBet.Id,
```

Comment:
This is not formatted correctly. You can run `go fmt ./...` from `calculator/` directory to format the whole application.